membership-common/src/main/scala/com/gu/services/model/PaymentDetails.scala

```diff
@@ -33,37 +32,35 @@ object PaymentDetails {
       nextPayment: Option[Payment],
       nextInvoiceDate: Option[LocalDate],
       lastPaymentDate: Option[LocalDate],
+      catalog: Catalog,
   ): PaymentDetails = {
 
     val firstPaymentDate = sub.firstPaymentDate
     val timeUntilPaying = Days.daysBetween(new LocalDate(DateTime.now()), new LocalDate(firstPaymentDate)).getDays
     import scala.math.BigDecimal.RoundingMode._
 
+    val plan = sub.plan(catalog)
     PaymentDetails(
       pendingCancellation = sub.isCancelled,
       startDate = sub.startDate,
-      chargedThroughDate = sub.plan.chargedThrough,
+      chargedThroughDate = plan.chargedThroughDate,
       customerAcceptanceDate = sub.acceptanceDate,
       nextPaymentPrice = nextPayment.map(p => (BigDecimal.decimal(p.price.amount) * 100).setScale(2, HALF_UP).intValue),
       lastPaymentDate = lastPaymentDate,
       nextPaymentDate = nextPayment.map(_.date),
       nextInvoiceDate = nextInvoiceDate,
       termEndDate = sub.termEndDate,
       paymentMethod = paymentMethod,
-      plan = PersonalPlan.paid(sub),
+      plan = PersonalPlan(
+        name = plan.productName,
+        price = plan.chargesPrice.prices.head,
+        interval = plan.billingPeriod.leftMap(e => throw new RuntimeException("no billing period: " + e)).toOption.get.noun,
```

Comment:
this used to be checked by the deserialisation stage, now we find out at this point so we deal with it as an exception